models.py

```diff
@@ -395,31 +404,30 @@ def friendly_status(self):
         return self.FRIENDLY_STATUSES.get(self.status)
 
     def send_assignment_notification(self, request, message, skip=False):
-        description = '{0} has been assigned as a proofreader for {1}'.format(
-            self.proofreader.full_name(),
-            self.round.article.title,
+
+        kwargs = {
+            'assignment': self,
+            'request': request,
+            'message': message,
+            'skip': skip,
+        }
+
+        events_logic.Events.raise_event(
+            plugin_settings.ON_PROOFREADER_ASSIGN_NOTIFICATION,
+            task_object=self.round.article,
+            **kwargs,
         )
 
         if not skip:
-            log_dict = {
-                'level': 'Info',
-                'action_text': description,
-                'types': 'Proofing Assignment',
-                'target': self.round.article,
-            }
-            notify_helpers.send_email_with_body_from_user(
-                request,
-                'Proofing Request',
-                self.proofreader.email,
-                message,
-                log_dict=log_dict,
-            )
-            notify_helpers.send_slack(
-                request,
-                description,
-                ['slack_editors'],
-            )
-
             self.notified = True
             self.save()
 
+    def send_cancel_notification(self, request):
+        description = '{0} cancelled a proofreading task for {article} assigned to {proofreader}'
+
+    def send_reset_notification(self, request):
+        pass
+
+    def send_complete_notification(self, request):
+        pass
+
```

Comment:
These require implementation still (consider the comments about the request argument)

Reply:
@mauromsl these were vestigial functions that are implemented elsewhere, deleted them now.